program/src/processor.rs

```diff
@@ -175,34 +198,223 @@ fn process_create_proposal(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
+        }
+
+        // Ensure the proposal account is not initialized.
+        if &proposal_info.try_borrow_data()?[0..8] != ArrayDiscriminator::UNINITIALIZED.as_slice() {
+            return Err(ProgramError::AccountAlreadyInitialized);
+        }
+
+        let clock = <Clock as Sysvar>::get()?;
+        let creation_timestamp = clock.unix_timestamp;
+
+        // Write the data.
+        let mut proposal_data = proposal_info.try_borrow_mut_data()?;
+        *bytemuck::try_from_bytes_mut::<Proposal>(&mut proposal_data)
+            .map_err(|_| ProgramError::InvalidAccountData)? = Proposal::new(
+            stake_authority_info.key,
+            creation_timestamp,
+            governance_config,
+        );
     }
 
-    // Ensure the proposal account has enough space.
-    if proposal_info.data_len() != std::mem::size_of::<Proposal>() {
-        return Err(ProgramError::InvalidAccountData);
+    // Initialize the proposal transaction account.
+    {
+        let (proposal_transaction_address, signer_bump_seed) =
+            get_proposal_transaction_address_and_bump_seed(proposal_info.key, program_id);
+        let bump_seed = [signer_bump_seed];
+        let proposal_transaction_signer_seeds =
+            collect_proposal_transaction_signer_seeds(proposal_info.key, &bump_seed);
+
+        // Ensure the provided proposal transaction address is the correct
+        // address derived from the program.
+        if !proposal_transaction_info
+            .key
+            .eq(&proposal_transaction_address)
+        {
+            return Err(PaladinGovernanceError::IncorrectProposalTransactionAddress.into());
+        }
+
+        // Ensure the proposal transaction account has not already been
+        // initialized.
+        if proposal_transaction_info.data_len() != 0 {
+            return Err(ProgramError::AccountAlreadyInitialized);
+        }
+
+        let state = ProposalTransaction::default();
+        let space = get_instance_packed_len(&state)? as u64;
+
+        // Allocate & assign.
+        invoke_signed(
+            &system_instruction::allocate(&proposal_transaction_address, space),
+            &[proposal_transaction_info.clone()],
+            &[&proposal_transaction_signer_seeds],
+        )?;
+        invoke_signed(
+            &system_instruction::assign(&proposal_transaction_address, program_id),
+            &[proposal_transaction_info.clone()],
+            &[&proposal_transaction_signer_seeds],
+        )?;
+
+        // Write the data.
+        state.serialize(&mut &mut proposal_transaction_info.try_borrow_mut_data()?[..])?;
     }
 
-    // Ensure the proposal account is not initialized.
-    if &proposal_info.try_borrow_data()?[0..8] != ArrayDiscriminator::UNINITIALIZED.as_slice() {
-        return Err(ProgramError::AccountAlreadyInitialized);
+    Ok(())
+}
+
+/// Processes a
+/// [InsertInstruction](enum.PaladinGovernanceInstruction.html)
+/// instruction.
+fn process_insert_instruction(
```

Comment:
nit: Since this is always appending at the end, and `insert` typically implies that it can be added anywhere, how about calling it `PushInstruction` or `AppendInstruction`?

Reply:
Yep that's fair. Going with push.